File: nyayasahayak-main-main/src/core/types/bench_memo.ts
// Bench Memo Types - Skill 08
// Mirrors backend/app/schemas/bench_memo.py

export type MemoStatus = 'draft' | 'generated' | 'reviewed';

export interface CasePrecedent {
    title: string;
    citation: string;
    summary: string;
    relevance: string;
}

export interface LegalIssue {
    issue: string;
    rule_of_law: string;
    analysis: string;
    conclusion: string;
}

export interface BenchMemo {
    id: string;
    case_id: string;
    judge_id: string;
    case_summary: string;
    procedural_history: string;
    legal_issues: LegalIssue[];
    precedents: CasePrecedent[];
    recommended_ruling: string;
    status: MemoStatus;
    created_at: string;
    updated_at: string;
    metadata: Record<string, any>;
}

export interface GenerateMemoRequest {
    case_id: string;
    focus_area?: string;
}

File: backend/app/schemas/bench_memo.py
"""
Bench Memo Generator Schemas - Skill 08
Auto-generates bench memos for judges
"""
from pydantic import BaseModel, Field
from typing import List, Optional, Dict, Any
from datetime import datetime
from enum import Enum


class MemoStatus(str, Enum):
    DRAFT = "draft"
    GENERATED = "generated"
    REVIEWED = "reviewed"


class CasePrecedent(BaseModel):
    title: str
    citation: str
    summary: str
    relevance: str


class LegalIssue(BaseModel):
    issue: str
    rule_of_law: str
    analysis: str
    conclusion: str


class BenchMemo(BaseModel):
    id: str
    case_id: str
    judge_id: str
    
    # Memo Content
    case_summary: str
    procedural_history: str
    legal_issues: List[LegalIssue]
    precedents: List[CasePrecedent]
    recommended_ruling: str
    
    status: MemoStatus
    created_at: datetime
    updated_at: datetime
    
    metadata: Dict[str, Any] = {}


class GenerateMemoRequest(BaseModel):
    case_id: str
    focus_area: Optional[str] = "General"  # e.g., "Bail", "Sentencing", "Admissibility"
